packages/x-charts/src/context/InteractionProvider.tsx

```diff
@@ -71,6 +93,9 @@ const dataReducer: React.Reducer<Omit<InteractionState, 'dispatch'>, Interaction
       return { ...prevState, item: null };
 
     case 'updateAxis':
+      if (action.data.x === prevState.axis.x && action.data.y === prevState.axis.y) {
```

Comment:
This update is not directly for voronoi. It's a way to "improve perf" If axis do not change, we do not create a new object to avoid triggering rerender at each mouse movement